src/argument_types.py

```diff
@@ -0,0 +1,206 @@
+from typing import ClassVar, Literal
+
+from pydantic import BaseModel, Field
+
+
+class Arguments(BaseModel):
+    """
+    Define arguments
+    """
+
+    lat: float = 0.0
+    long: float = 0.0
+    city: str = "pleasure_point"
+    show_wave: bool = True
+    show_large_wave: bool = False
+    show_uv: bool = True
+    show_height: bool = True
+    show_direction: bool = True
+    show_period: bool = True
+    show_city: bool = True
+    show_date: bool = True
+    show_air_temp: bool = False
+    show_wind_speed: bool = False
+    show_wind_direction: bool = False
+    json_output: bool = False
+    show_rain_sum: bool = False
+    show_precipitation_prob: bool = False
+    unit: Literal["imperial", "metric"] = "imperial"
+    decimal: int = Field(default=1, ge=0)
+    forecast_days: int = Field(default=0, ge=0, le=7)
+    color: str = "blue"
+    gpt: bool = False
+
+
+class ArgumentMappings(BaseModel):
```

Comment:
I think the implementation is as I expected, so it looks good!

If I may add a suggestion.
The fields in the `ArgumentMappings` class overlap with those in the `Arguments` class. Ideally, if possible, I think it would be better to consolidate all arguments into the `Arguments` class and create the final mapping dictionary based on the contents of the `Arguments` class.(For example, when implementing new arguments in the future, we would likely need to add fields to both classes, resulting in dual management, which could lead to reduced maintainability.)

However, I believe that realizing this would require modifications to the `cli.py` file as well, which could potentially be a substantial amount of work. Therefore, I'll leave the decision on whether to implement this up to you :)
It's perfectly fine to keep this as a future refactoring phase if you prefer!

@ryansurf  How do you think?

Reply:
Agreed, it would be ideal to reduce any redundancy.

`cli.py` would indeed need to be revamped. Let us know what you want to do @avkoll !